ProofWidgets/Demos/Euclidean.lean

```diff
@@ -0,0 +1,177 @@
+/-
+ Copyright (c) 2023 Vladimir Sedlacek. All rights reserved.
+ Released under Apache 2.0 license as described in the file LICENSE.
+ Authors: Vladimir Sedlacek
+ -/
+
+import Lean.Data.HashMap
+import Lean.Elab.Tactic
+import ProofWidgets.Component.PenroseDiagram
+import ProofWidgets.Component.HtmlDisplay
+import ProofWidgets.Component.Panel
+
+open Lean Meta Server
+open ProofWidgets
+
+/-! # Minimal definitions of synthetic geometric primitives, inspired by https://github.com/ah1112/synthetic_euclid_4 -/
+
+universe u1 u2 u3
+class incidence_geometry :=
```

Comment:
Since this would be demo code that people may copy, let's stick to [Lean 4 style conventions](https://github.com/leanprover/lean4/blob/808bb9b579513b47d0e87cb7ff7aade705a71d79/doc/lean3changes.md#style-changes). For example `IncidenceGeometry`, `Point : Type u1`, etc.